src/frontend/src/flows/authenticate/index.ts

```diff
@@ -24,72 +26,106 @@ import {
+      <!-- The title is hidden but used for accessibility -->
+      <h1 class="is-hidden">Internet Identity</h1>
+      <div class="c-logo">${icLogo}</div>
+      <div class="l-stack">
+        <p class="t-lead" style="text-align: center;">
+          Connect to<br />
+          <span class="t-strong">${hostName}</span>
+          ${derivationOrigin !== undefined && derivationOrigin !== hostName
+            ? html` <span class="c-tooltip t-link__icon"
+                >${attentionIcon}<i
+                  class="c-tooltip__message c-card c-card--narrow"
+                  >An alternative origin of ${derivationOrigin}</i
+                ></span
+              >`
+            : ""}
+          <br />
+        </p>
+      </div>
+
+      <div class="l-stack c-input c-input--vip c-input--anchor">
+        <input
+          ${ref(userNumberInput)}
+          type="text"
+          id="userNumberInput"
+          placeholder="Enter anchor"
+          value="${userNumber !== undefined ? userNumber : ""}"
+          style="width: 100%;"
+        />
+
+        <p
+          id="invalidAnchorMessage"
+          class="anchor-error-message is-hidden t-paragraph t-strong"
+        >
+          The Identity Anchor is not valid. Please try again.
+        </p>
+      </div>
 
-    <button id="authorizeButton" class="c-button">Authorize</button>
-    <div class="l-stack">
-      <ul class="c-list--flex">
-        <li>
-          <a id="registerButton" class="t-link">Create Anchor</a>
-        </li>
-        <li>
-          <a id="recoverButton" class="t-link">Lost Access?</a>
-        </li>
-        <li>
-          <a class="t-link" id="manageButton">Manage Anchor</a>
-        </li>
-        <li>
-          <a
-            class="t-link"
-            href="/faq"
-            target="_blank"
-            rel="noopener noreferrer"
-            >FAQ</a
-          >
-        </li>
-      </ul>
+      <button ${ref(authorizeButton)} id="authorizeButton" class="c-button">
+        Authorize
+      </button>
+      <div class="l-stack">
+        <ul class="c-list--flex">
+          <li>
+            <a ${ref(registerButton)} id="registerButton" class="t-link"
+              >Create Anchor</a
+            >
+          </li>
+          <li>
+            <a @click="${onRecoverClick}" id="recoverButton" class="t-link"
+              >Lost Access?</a
+            >
+          </li>
+          <li>
+            <a @click="${onManageClick}" class="t-link" id="manageButton"
+              >Manage Anchor</a
+            >
+          </li>
+          <li>
+            <a
+              class="t-link"
+              href="/faq"
+              target="_blank"
+              rel="noopener noreferrer"
+              >FAQ</a
+            >
+          </li>
+        </ul>
+      </div>
     </div>
-  </div>
-  ${footer}`;
+    ${footer}`;
+
+  return {
+    template,
+    keys: { authorizeButton, userNumberInput, registerButton },
```

Comment:
is the reference static or dynamic?

e.g. if a consumer renders the template and gets a ref, if that particular ref gets dynamically removed within the component, is the consumer informed the ref is not valid anymore?

I mean something like the React `forwardRef` pattern.

Reply:
The actual ref is dynamic, but after it's been extracted (in `renderTemplateRef`, with `.value`) then it's static. It's not ideal, but the behavior is similar to what we'd have before with `document.getElementById("...")` where the returned elements may not be attached to the DOM forever.

Reply:
I see. Do you actually need to expose these ref or can you bind function / method to the template? If yes would it be cleaner to do so?

```
@Stencil
@tag 'my-yolo'
export class YoloComponent {

  ref!: HTMLDivElement;
  
  @Method()
  doSomething() {
      this.ref.letsDoIt();
  }

   render <div ref=el />
}

<html>
   <script>
      document.querySelector('my-yolo').doSomething();
   </script>
   
   <my-yolo />
</html>
```

Honestly just speaking out at loud since you said it's a bit an experiment. Totally fine with ref.

Reply:
Mh, the template itself isn't a component, just (basically) a string to be rendered. Are you suggesting bypassing lit's `render`? I'm not sure I understand

Reply:
Ah alright, somehow the custom element discussion sticked in my head. So all clear.